### PR TITLE
Documentation refactor + nitpicky + example review + chaos

### PR DESCRIPTION
**Pull request by nabobalis**
Aim to add more info about using aiapy and reduce any overlap with the sunpy development documentation.

TODO:
- [ ] Work out what could be missing and fix it.

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -1,36 +1,76 @@
...
+
+===================
 aiapy documentation
 ===================
 
-``aiapy`` is a Python package for analyzing data from the Atmospheric Imaging Assembly (AIA) instrument onboard the Solar Dynamics Observatory (SDO) spacecraft.
+``aiapy`` is a Python library for analyzing data from the Atmospheric Imaging Assembly (AIA) instrument onboard the Solar Dynamics Observatory (SDO) spacecraft.
 
-``aiapy`` includes software for converting AIA images from level 1 to level 1.5, point spread function deconvolution, and computing the wavelength and temperature response functions for the EUV channels.
+It allows for converting AIA images from level 1 to level 1.5, point spread function deconvolution, and computing the wavelength and temperature response functions for the EUV channels.
```
Comment (wtbarnes) [inline docs/index.rst:9]:
```suggestion
It allows for converting AIA images from level 1 to level 1.5, point spread function deconvolution, and computing the wavelength response functions for the EUV channels.
```
Can't compute the temperature responses yet

File: docs/topic_guide/installation.rst [hunk 2]
```diff
@@ -0,0 +1,102 @@
+.. _aiapy-topic-guide-installing:
+
+************
+Installation
```
Comment (wtbarnes) [inline docs/topic_guide/installation.rst:4]: Why is this repeated. Shouldn't we just direct people to the SunPy installation guide, especially for things like installing Python?

File: examples/instrument_degradation.py [hunk 3]
```diff
@@ -37,37 +37,46 @@
...
-deg = {c: degradation(c, time, correction_table=correction_table) for c in channels}
+deg = {channel: degradation(channel, time_range, correction_table=correction_table) for channel in aia_channels}
 
-###########################################################
+###############################################################################
 # Plotting the different degradation curves as a function of time, we can
 # easily visualize how the different channels have degraded over time.
-time_support(format="jyear")  # This lets you pass astropy.time.Time objects directly to matplotlib
+
+# This lets you pass astropy.time.Time objects directly to matplotlib
```
Comment (wtbarnes) [inline examples/instrument_degradation.py:68]:
```suggestion
# This lets you pass `astropy.time.Time` objects directly to matplotlib
```

File: examples/instrument_degradation.py [hunk 4]
```diff
@@ -37,37 +37,46 @@
...
+time_support(format="jyear")
+
 fig = plt.figure()
 ax = fig.gca()
-for c in channels:
-    ax.plot(time, deg[c], label=f"{c.value:.0f} Å")
-ax.set_xlim(time[[0, -1]])
+
+for channel in aia_channels:
+    ax.plot(time_range, deg[channel], label=f"{channel.value:.0f} Å")
```
Comment (wtbarnes) [inline examples/instrument_degradation.py:75]:
```suggestion
    ax.plot(time_range, deg[channel], label=f"{channel:latex}")
```

File: examples/skip_correct_degradation.py [hunk 5]
```diff
@@ -15,53 +14,57 @@
...
 # First, let's fetch some metadata for the 335 Å channel of AIA between 2010
-# and 2018 at a cadence of 30 days. We choose the 335 Å channel because it has experienced
+# and 2022 at a cadence of 180 days. We choose the 335 Å channel because it has experienced
 # significant degradation compared to the other EUV channels.
+
 results = Fido.search(
-    a.Time("2010-06-01T00:00:00", "2021-06-01T00:00:00"),
-    a.Sample(30 * u.day),
+    a.Time("2010-06-01T00:00:00", "2022-01-01T00:00:00"),
+    a.Sample(180 * u.day),
```
Comment (wtbarnes) [inline examples/skip_correct_degradation.py:29]: Does it pass at this cadence?

File: examples/skip_correct_degradation.py [hunk 6]
```diff
@@ -15,53 +14,57 @@
...
 
-###########################################################
+###############################################################################
 # To understand the effect of the degradation and the correction factor, we
 # plot the corrected and uncorrected mean intensity as a function of time.
 # Note that the uncorrected intensity decreases monotonically over time
 # while the corrected intensity recovers to pre-2011 values in 2020.
+
+# These are needed to allow the use of astropy quantities and
+# time objects in the plot.
```
Comment (wtbarnes) [inline examples/skip_correct_degradation.py:64]: We already said this in one gallery entry. Does it need to be in both? Also, should we move all of these to the top of the file? I cannot remember what we do in sunpy.

File: docs/index.rst [hunk 7]
```diff
@@ -1,36 +1,76 @@
...
+    :gutter: 2
+
+    .. grid-item-card:: Topic Guide
+        :link: aiapy-topic-guide-index
+        :link-type: ref
+        :text-align: center
+
+        :material-outlined:`accessibility_new;8em;sd-text-secondary`
+
+        **New users start here!**
```
Comment (wtbarnes) [inline docs/index.rst:21]: Topic guides are not for new users
```suggestion
```

File: docs/topic_guide/index.rst [hunk 8]
```diff
@@ -0,0 +1,14 @@
+.. _aiapy-topic-guide-index:
+
+*****************
+aiapy topic guide
```
Comment (wtbarnes) [inline docs/topic_guide/index.rst:4]: I'm not really sure either of these are topic guides. Maybe the data prep one could be with some tweaking but I don't think it's worth having a whole section of the docs just for a single entry.

**Review by wtbarnes — CHANGES_REQUESTED** [review]
Requesting changes because I think the docs need some reorganizing.

File: docs/topic_guide/installation.rst (hunk 2 above)
Comment (nabobalis) [reply docs/topic_guide/installation.rst:4]: Duplicated to keep the flow of the docs more internal but I can do this.

File: examples/skip_correct_degradation.py (hunk 5 above)
Comment (nabobalis) [reply examples/skip_correct_degradation.py:29]: It did on my laptop this morning.

File: examples/skip_correct_degradation.py (hunk 6 above)
Comment (nabobalis) [reply examples/skip_correct_degradation.py:64]: Will do.

File: docs/topic_guide/index.rst (hunk 8 above)
Comment (nabobalis) [reply docs/topic_guide/index.rst:4]: Sure. I will think again.

File: examples/skip_correct_degradation.py (hunk 6 above)
Comment (nabobalis) [reply examples/skip_correct_degradation.py]: I decided to leave them separate but reduce the text in the correcting example.

Also doing a smaller range now means it returns.

File: examples/skip_correct_degradation.py (hunk 5 above)
Comment (wtbarnes) [reply examples/skip_correct_degradation.py]: Oh nice! Can we un-skip this now then?

Comment (nabobalis) [reply examples/skip_correct_degradation.py]: Aready done

**Comment by nabobalis** [comment]
This PR got really out of hand.

File: docs/getting_started.rst [hunk 9]
```diff
@@ -1,24 +1,13 @@
...
+******************************
+Getting started with ``aiapy``
+******************************
 
-.. code-block:: shell
+``aiapy`` is a community-developed, free and open-source library to to convert AIA images from level 1 to level 1.5, compute point spread function deconvolution, and computing the wavelength response functions for the EUV channels.
 
-   pip install aiapy
+.. toctree::
+    :maxdepth: 2
```
Comment (nabobalis) [inline docs/getting_started.rst:10]:
```suggestion
    :maxdepth: 1
```

File: docs/getting_started.rst [hunk 10]
```diff
@@ -1,24 +1,13 @@
...
-===============
+.. _aiapy-getting-started-index:
 
-The stable version of ``aiapy`` is available through both PyPI via ``pip``,
+******************************
+Getting started with ``aiapy``
+******************************
 
-.. code-block:: shell
+``aiapy`` is a community-developed, free and open-source library to to convert AIA images from level 1 to level 1.5, compute point spread function deconvolution, and computing the wavelength response functions for the EUV channels.
```
Comment (wtbarnes) [inline docs/getting_started.rst:7]:
```suggestion
``aiapy`` is a community-developed, free and open-source Python library for analyzing data from the Atmospheric Imaging Assembly (AIA) onboard the Solar Dynamics Observatory (SDO). 
Capabilities include converting AIA images from level 1 to level 1.5, deconvolving images with the instrument point spread function, and computing the wavelength response functions for the EUV channels.
```

Comment (wtbarnes) [reply docs/getting_started.rst:7]: Actually, maybe this text can just be removed altogether. There's not really a need to repeat what's on the main page.

File: docs/installation.rst [hunk 11]
```diff
@@ -0,0 +1,102 @@
+.. _aiapy-installing:
```
Comment (wtbarnes) [inline docs/installation.rst:1]: As I've said, I don't like having an exact copy of what is already in the sunpy docs. I think this page should just point to the sunpy install page.

File: docs/preparing_data.rst [hunk 12]
```diff
@@ -1,16 +1,21 @@
...
 
-AIA data provided by the JSOC are level 1 data products.
-This means that the images still include the roll angle of the satellite and
-each channel may have a slightly different resolution.
+============================================
+Preparing AIA data from level 1 to level 1.5
+============================================
+
+AIA data products provided by the JSOC are level 1 data products.
+This means that the images still include the roll angle of the satellite and each channel may have a slightly pixel scale.
```
Comment (wtbarnes) [inline docs/preparing_data.rst:8]:
```suggestion
This means that the images still include the roll angle of the satellite and each channel may have a slightly different pixel scale.
```

File: docs/preparing_data.rst [hunk 13]
```diff
@@ -1,16 +1,21 @@
...
+This is important if you want to compare images from different channels or create Differential Emission Measure (DEM) maps.
+
+The promotion to level 1.5 involves updating the pointing keywords, removing the roll angle, scaling the image to a resolution of 0.6 arcseconds per pixel, and translating the image such that the center of the Sun is located in the center of the image.
 
-In IDL, this is done with the `aia_prep.pro` procedure in SSWIDL as described in the `SDO Analysis Guide <https://www.lmsal.com/sdodocs/doc/dcur/SDOD0060.zip/zip/entry/index.html>`__.
+In IDL, this is done with the ``aia_prep.pro`` procedure in SSWIDL as described in the `SDO Analysis Guide <https://www.lmsal.com/sdodocs/doc/dcur/SDOD0060.zip/zip/entry/index.html>`__.
 The following example, :ref:`sphx_glr_generated_gallery_prepping_level_1_data.py` demonstrates how to achieve this in Python with ``aiapy``.
 
-Additional data processing steps (e.g. a PSF deconvolution) should be done in the following order:
+However, this is not the entire possible processing steps that can be done to AIA level 1 data products.
```
Comment (wtbarnes) [inline docs/preparing_data.rst:17]:
```suggestion
There are also additional processing steps that can be applied to the level 1 AIA images.
```

**Review by wtbarnes — CHANGES_REQUESTED** [review]

File: docs/getting_started.rst (hunk 10 above)
Comment (nabobalis) [reply docs/getting_started.rst:7]: I can remove it.

File: docs/installation.rst (hunk 11 above)
Comment (nabobalis) [reply docs/installation.rst:1]: Tweaked.

**Review by wtbarnes — APPROVED** [review]

File: aiapy/data/sample.py [hunk 14]
```diff
@@ -1,33 +1,79 @@
 """
-This module provides the following sample data files. These files are
-downloaded when this module is imported for the first time.
+This module provides the following sample data files.
 
+When a sample shortname is accessed, the corresponding file is downloaded if needed.
+All files can be downloaded by calling :func:`~irispy.data.sample.download_all`.
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply aiapy/data/sample.py:5]: DAMN IT

File: aiapy/version.py [hunk 15]
```diff
@@ -1,18 +1,23 @@
 # NOTE: First try _dev.scm_version if it exists and setuptools_scm is installed
-# This file is not included in wheels/tarballs, so otherwise it will
+# This file is not included in irispy-lmsal wheels/tarballs, so otherwise it will
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply aiapy/version.py:2]: DAMN IT

File: aiapy/calibrate/meta.py [hunk 16]
```diff
@@ -116,10 +136,13 @@ def update_pointing(smap, *, pointing_table=None):
...
     if not t_obs_in_interval.any():
-        raise IndexError(
+        msg = (
             f"No valid entries for {t_obs} in pointing table "
             f'with first T_START date of {pointing_table[0]["T_START"]} '
-            f'and a last T_STOP date of {pointing_table[-1]["T_STOP"]}.',
+            f'and a last T_STOP date of {pointing_table[-1]["T_STOP"]}.'
+        )
+        raise IndexError(
+            msg,
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply aiapy/calibrate/meta.py:145]: You do not like start of the art code formatting?

[the remark this answers is not in the document]
Comment (nabobalis) [reply aiapy/calibrate/meta.py]: Fixed now

**Pull request merged**